src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -725,8 +725,7 @@ private void computeUpstreamNodesWhenWorkspaceModified(WorkspacesModificationEve
             if (args.Id == this.Guid)
             {
                 this.workspaceLoaded = true;
-                var frozenNodes = nodes.Where(x => x.isFrozenExplicitly).ToList();
-                frozenNodes.ForEach(z => z.ComputeUpstreamOnDownstreamNodes());
+                this.ComputeUpstreamCacheForEntireGraph();
```

Comment:
the way I have implemented this is faster that what we were doing before since even if the graph contained 2 frozen nodes that had overlapping subtrees that would be recomputed.In the worst case the running time of this entire function should be O(n) where the other would could potentially be O(n^2) - imagine a graph where many nodes had been explicitly frozen.